source/MRMesh/MRColorMapAggregator.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+#include "MRMeshFwd.h"
+#include "MRVector.h"
+#include "MRColor.h"
+#include "MRId.h"
+#include "MRBitSet.h"
+
+
+namespace MR
+{
+/**
+ * @brief Class for aggregate several color map in one
+ * @detail Color map are aggregated according order
+ */
+template<typename Tag>
+class ColorMapAggregator
+{
+public:
+    using ColorMap = Vector<Color, Id<Tag>>;
+    using ElementBitSet = TaggedBitSet<Tag>;
+
+    ColorMapAggregator() = default;
+
+    /// set default (background) color
+    MRMESH_API void setDefaultColor( const Color& color );
+
+    /**
+     * @brief add color map after all (more priority)
+     * @param colorMap color map for elements
+     * @param elementBitSet bitset of elements for which the color map is applied
+     */
+    MRMESH_API void pushBack( const ColorMap& colorMap, const ElementBitSet& elementBitSet );
+
+    /**
+     * @brief insert color map after element #i (0 - minimum priority)
+     * @param colorMap color map for elements
+     * @param elementBitSet bitset of elements for which the color map is applied
+     */
+    MRMESH_API void insert( int i, const ColorMap& colorMap, const ElementBitSet& elementBitSet );
+
+    /**
+     * @brief replace color map in #i position
+     * @param colorMap color map for elements
+     * @param elementBitSet bitset of elements for which the color map is applied
+     */
+    MRMESH_API void replace( int i, const ColorMap& colorMap, const ElementBitSet& elementBitSet );
+
+    /**
+     * @brief erase n color map from #i 
+     */
+    MRMESH_API void erase( int i, int n = 1 );
+
+    /// color map aggregating mode
+    enum class AggregateMode
+    {
+        Overlay, /// result element color is element color of more priority color map (or default color, if there isn't color map for this element)
+        Blending /// result element color is blending colors of all color map in this element and default color (https://en.wikipedia.org/wiki/Alpha_compositing)
+    };
+
+    /// set color map aggregating mode
+    MRMESH_API void setMode( AggregateMode mode );
+
+    /// get aggregated color map
+    MRMESH_API ColorMap aggregate();
```

Comment:
aggregate should accept object valid ElementBitSet, to aggregate color map for it, no need to store `colorMapSize_`

Reply:
reworked